app/code/core/Mage/Adminhtml/Block/Dashboard/Graph.php

```diff
@@ -215,7 +215,7 @@ public function getChartUrl($directUrl = true)
                     $dateStart->addMonth(1);
                     break;
             }
-            foreach ($this->getAllSeries() as $index => $serie) {
+            foreach (array_keys($this->getAllSeries()) as $index) {
```

Comment:
Why change the code like this? It seems less readable and possibly less performant.

Reply:
what about this?
https://stackoverflow.com/a/29556153 

For me, is a questionable choice

If it is among the rules of Rector, it is not obvious that we have to use it. It seemed more sensible to me to approve the short syntax for the echo <?=.

Reply:
I know that post at SE ... it looks like overhead, but "it's not always a bad thing if a code generates more opcode ".

I'd prefer it. No exclude rules for phpmd/rector ... Finally its about personal preferences.

Short echo syntax ... this would change so many places that are referenced in docs/posts (like at SE) for only to save some bytes for no benefit.

Reply:
seems to me an example of "Premature optimization is the root of all evil."

short syntax is not for save bytes but readabily and esthetics

also another rule that adds time to the very long pipeline that I have to wait for every PR

Reply:
The discussion about the syntax is off-topic. It can be discussed elsewhere, it was proposed in a PR and I did not agree with it. In my opinion, the whole template  inherited from Magento is not readable at all, if I compare for example with Twig.

Reply:
Twig is also off-topic, but an interssting idea. :P

> seems to me an example of "Premature optimization is the root of all evil."

Well. I like clean code. Having variables that are not used looks not clean (to me).

Reply:
This means that every time I have a foreach where I use only the keys, I have to use array_keys too.
This results in two internal loops of the same array.
What do others think about this? @colinmollenhour @kiatng @luigifab @drobinson @LeeSaferite @elidrissidev @tmotyl ?

Reply:
Performance isnt an issue. And how ofter does it happen with your own code?

Reply:
I say if i have to commit new code after this PR i have to follow always this standard.
About performance it results into 2 loops of the same array multiplied by every instance where the code occurs

Reply:
Dont care about that two loops. Really.

You dont need to follow this standard. At the end there should be a working rector-config that changes it. (Working on it, but I wawnt to apply the rules step by step, because PRs would become to large)

Reply:
@empiricompany i am a fan of micro-optimization, but here I'd prefer "cleaner" code over it. 

An o/c rector rule can be disabled. Same for phpmd.

Reply:
ok, but this for me is not cleaner it's a horror

Reply:
My 2 cents: I prefer just leaving the code alone unless there is noticeable performance or readability improvement in the new one, both of wish I don't see here.

Reply:
Using Rector is fine.

Now it depends on who writes the code, if he is a young programmer he will use the version proposed by Sven, which he is already used to. If he is an older programmer, it will be difficult for him to adapt to the new code. 

As for me, I think I have to adapt.

Reply:
Btw ... the the people behind rector are highly active. My last issue was fixed within 15 minutes and a new release including it was made today.

phpstan and rector etc.  are tools that are used, supported, sponsored by far far bigger projects ... why? It makes code better. Standardizes it.

We should use it too.

Reply:
Soryy but I can't see this as a standard and I never seen it in other major framework

Reply:
Not a major one ....

https://github.com/search?q=repo%3Adrupal%2Fdrupal%20%22foreach%20(array_keys%22&type=code

Reply:
Oh! I would have never expected that!

https://github.com/search?q=repo%3Alaravel%2Fframework+foreach+%28array_keys&type=code
https://github.com/search?q=repo%3Asymfony%2Fsymfony%20foreach%20(array_keys&type=code

Even though I still don't like it, and I believe they are also victims of these absurd rules, you have my approval.

Reply:
At the end of the day ... how often do you check that source-code?  Let it fix  and follow "common" rules. ;)

Reply:
Drupal is not a CMS for everyone, the cream of the programmers gathered here and I always found solutions from which to draw inspiration. Many, many years ago I had the choice to study between Joomla and Drupal and I stayed attached to Drupal, even though it's damn hard because they only use what is new and you can't really keep up with the developers.

Reply:
> Even though I still don't like it

;) 

I think all, ALL big projects use reactor/phpstan/phpcs to ensure some code quality.  Maybe it looks strange whem you have never used it, it makes sense to me. Readablity come form how often yo read it :P

Reply:
@ADDISON74 you cant compare Drupal or Typo3 with Joomla or Wordpress .... thats another level.

Reply:
I dislike it, I prefer `foreach ($things as $key = $nul) ...`

I'm also using Rector and PhpStorm ins pect code when I'm not traveling, with "groups enabled", but I disable some rules.

Reply:
I thought it would be a rule everybody agrees with ... :P